### DISCUSSION TITLE
Segment vs Tot Seg and other columns

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [DaxStudio/DaxStudio#1042] [General]
What is the number shown in the Tot Seg column? In my data it is about 14x of the Segment number

![2023-02-02_15-30-31](https://user-images.githubusercontent.com/73054674/216294253-bee7c964-ef47-4115-8dd3-dc988e5363d3.png)

Also, can you please share some details about the other columns - Pageable, Resident, Temperature. Thanks!

**Comment by dgosbell** [comment]
**Segments** - is the maxium number of segments in any single column in that partition (so usually the segments in the column with the highest cardinality)

**Tot Seg.** - is the total number of segments in that partitions for all columns 

The other columns come from this DMV and are documented there [DISCOVER_STORAGE_TABLE_COLUMN_SEGMENTS](https://learn.microsoft.com/en-us/openspecs/sql_server_protocols/ms-ssas/6f631e7e-9863-4d5a-8254-e7603efe271c)

**Reply by SharmaAntriksh** [reply]
Thanks! The definition of **_Segments_** you have shared is something new, I know that data in a table is stored and compressed by segments for all columns (not per column) and each partition can have many segments, but I never knew that a segment has a direct dependency on the column cardinality.

Do you have any example of why/when the Tot Seg information is useful?

**Reply by marcosqlbi** [reply]
I think we covered a complete description of the VertiPaq engine in some of our books.
Because a segment is the minimum allocation unit for parallelism (one segment can be scanned by only one core), that information gives you an overall idea of the allocation/fragmentation cost of the table and the possible parallelism. However, it's not much helpful per se.

**Reply by SharmaAntriksh** [reply]
What about the "_Segments - is the maxium number of segments in any single column in that partition_" does this mean segments are both at column and table level?

**Reply by marcosqlbi** [reply]
Yes, usually it's simmetric, all the columns have the same segments. But because we read a DMV, we provide the MAX value only (if there was a column with a smaller number of segments you don't see that until you drill-down at column details.

**Reply by frithjofv** [reply]
It seems to me that the Total segments in a table is equal to Segments x (Column count + 1), where Segments is the number of segments in a column.

For example, I have a table with 50 Segments and 10 Columns, the Total segments is 550. 

I'm wondering why Total segments depend on Column count + 1, why not just Column count?

It seems that Pageable = Segments x Column count.

**Reply by marcosqlbi** [reply]
Yes you are right, currently it is the sum of all the segments in all the columns (the additional column is the internal row number).
